gensim/models/phrases.py

```diff
@@ -177,6 +176,13 @@ def __init__(self, sentences=None, min_count=5, threshold=10.0,
         # set scoring based on string
         # intentially override the value of the scoring parameter rather than set self.scoring here,
         # to still run the check of scoring function parameters in the next code block
+
+        # for python 2 and 3 compatibility. basestring is used to check if scoring is a string
```

Comment:
Almost there :) We use `six` in gensim for py2/py3 compatibility, so `isinstance` on `six.string_types` is probably what we want.